From stable+bounces-114135-greg=kroah.com@vger.kernel.org Thu Feb  6 17:23:08 2025
From: Koichiro Den <koichiro.den@canonical.com>
Date: Fri,  7 Feb 2025 01:22:16 +0900
Subject: Revert "btrfs: avoid monopolizing a core when activating a swap file"
To: gregkh@linuxfoundation.org, stable@vger.kernel.org
Cc: wqu@suse.com, fdmanana@suse.com, dsterba@suse.com
Message-ID: <20250206162217.1387360-1-koichiro.den@canonical.com>

From: Koichiro Den <koichiro.den@canonical.com>

This reverts commit 9f372e86b9bd1914df58c8f6e30939b7a224c6b0.

The backport for linux-6.12.y, commit 9f372e86b9bd ("btrfs: avoid
monopolizing a core when activating a swap file"), inserted
cond_resched() in the wrong location.

Revert it now; a subsequent commit will re-backport the original patch.

Fixes: 9f372e86b9bd ("btrfs: avoid monopolizing a core when activating a swap file") # linux-6.12.y
Signed-off-by: Koichiro Den <koichiro.den@canonical.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 fs/btrfs/inode.c |    2 --
 1 file changed, 2 deletions(-)

--- a/fs/btrfs/inode.c
+++ b/fs/btrfs/inode.c
@@ -7200,8 +7200,6 @@ noinline int can_nocow_extent(struct ino
 			ret = -EAGAIN;
 			goto out;
 		}
-
-		cond_resched();
 	}
 
 	if (file_extent)
